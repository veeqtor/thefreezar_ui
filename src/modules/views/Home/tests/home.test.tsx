// /* eslint-disable @typescript-eslint/explicit-function-return-type */
// /* eslint-disable @typescript-eslint/no-explicit-any */

// import * as React from 'react';
// import { render, cleanup, fireEvent } from '@testing-library/react';
// import { Provider } from 'react-redux';

// // import HomePage from 'modules/views/Home';
// import * as navigationActions from 'store/actions/navigation';
// import configureStore from 'store/config';
// import { lazy } from '@loadable/component';
// import withSuspense, { IWithSuspenseProps } from 'modules/HOC/withSuspense';

// jest.mock('store/actions/navigation');

// const renderHomePage = (store: any, props: any) => {
//   const LazyComponent = lazy(() => import(`modules/views/Home`));

//   const HomePage = withSuspense({ page: 'Home', ...props });
//   return {
//     ...render(
//       <Provider store={store}>
//         <React.Suspense fallback="loading">
//           <LazyComponent {...props} />
//         </React.Suspense>
//       </Provider>,
//     ),
//   };
// };

// describe('HomeContainer Test', () => {
//   let store: any;
//   const mockedgoToNextPage = navigationActions.goToNextPage as jest.Mock;
//   const mockedretrieveLastPageState = navigationActions.retrieveLastPageState as jest.Mock;

//   beforeAll(() => {
//     mockedgoToNextPage.mockImplementation(() => ({ type: '' }));
//     mockedretrieveLastPageState.mockImplementation(() => ({ type: '' }));
//   });

//   beforeEach(() => {
//     store = configureStore('test');
//     mockedretrieveLastPageState.mockClear();
//     mockedgoToNextPage.mockClear();
//   });

//   afterEach(cleanup);

//   it('Should render welcome text', () => {
//     const {
//       getByText,
//       container: { firstChild },
//     } = renderHomePage(store, { title: 'Alejandro' });
//     expect(getByText('Welcome Home!! Alejandro')).toBeInTheDocument();
//     expect(firstChild).toMatchSnapshot();
//   });

//   it('Should call the redux dispatches', () => {
//     const { getByText } = renderHomePage(store, { title: 'Alejandro' });
//     fireEvent.click(getByText('Click me'));
//     expect(navigationActions.goToNextPage).toHaveBeenCalled;
//     expect(navigationActions.retrieveLastPageState).toHaveBeenCalled;
//   });
// });

export default {};
