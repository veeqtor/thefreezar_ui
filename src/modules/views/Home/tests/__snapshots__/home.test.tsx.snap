// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeContainer Test Should render welcome text 1`] = `
<section
  class="css-1mldeik"
>
  <div
    class="css-13rb3nt"
  />
</section>
`;
